src/components/Notification/Notification.tsx

```diff
@@ -0,0 +1,175 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import React, { useEffect, useRef } from "react";
+import type { HTMLProps, ReactNode } from "react";
+
+import Button, { ButtonAppearance } from "../Button";
+import type { Props as ButtonProps } from "../Button";
+
+export const notificationTypes = {
+  CAUTION: "caution",
+  INFORMATION: "information",
+  NEGATIVE: "negative",
+  POSITIVE: "positive",
+};
+
+/**
+ * The props for the Notification component.
+ */
+export type Props = {
+  /**
+   * The list of actions that the notification can perform.
+   */
+  actions?: {
+    appearance?: ButtonProps["appearance"];
+    className?: string;
+    label: string;
+    onClick: () => void;
+  }[];
+  /**
+   * Whether the notification should not show a border.
+   */
+  appearance?: "borderless" | "default" | "modal" | "raised";
+  /**
+   * The notification message content.
+   */
+  children?: ReactNode;
+  /**
+   * Optional class(es) to apply to the parent notification element.
+   */
+  className?: string;
+  /**
+   * Whether the title should display inline with the message.
+   */
+  inline?: boolean;
+  /**
+   * The function to run when dismissing/closing the notification.
+   */
+  onDismiss?: () => void;
+  /**
+   * The severity of the notification.
+   */
+  severity?: "caution" | "information" | "negative" | "positive";
```

Comment:
Could this be a TypeScript enum?

Reply:
It could be yes, but I chose not to so that you don't have to also import the enum if you want to use the Notification component in a TypeScript project. i.e. with an enum it would have to be `severity={NotificationSeverity.CAUTION}` and could not be `severity="caution"` even though they evaluate to the same string. The React Typescript cheatsheet actually [recommends avoiding enums](https://react-typescript-cheatsheet.netlify.app/docs/basic/troubleshooting/types/#enum-types), surprisingly.

Anyway I like how enums mean you only define the string in one place, so as a compromise I've created defined constants (`NotificationAppearance` and `NotificationSeverity`) that you can import if you want, otherwise the string will work fine too.

Reply:
Thanks for clarifications.